src/main/java/slimeknights/tconstruct/tables/client/inventory/module/TinkerStationButtonsWidget.java

```diff
@@ -0,0 +1,77 @@
+package slimeknights.tconstruct.tables.client.inventory.module;
+
+import net.minecraft.client.gui.components.Button;
+import slimeknights.tconstruct.library.client.Icons;
+import slimeknights.tconstruct.library.tools.layout.StationSlotLayout;
+import slimeknights.tconstruct.library.tools.layout.StationSlotLayoutLoader;
+import slimeknights.tconstruct.tables.client.inventory.TinkerStationScreen;
+import slimeknights.tconstruct.tables.client.inventory.widget.SlotButtonItem;
+
+import java.util.List;
+
+public class TinkerStationButtonsWidget extends SideButtonsWidget<SlotButtonItem> {
+
+  private int style = 0;
+
+  public static final int WOOD_STYLE = 2;
+  public static final int METAL_STYLE = 1;
+
+  public TinkerStationButtonsWidget(TinkerStationScreen parent) {
+    super(parent, TinkerStationScreen.COLUMN_COUNT, false);
+
+    // Logic to run when a button is pressed
+    Button.OnPress onButtonPressed = self -> {
+      for (SlotButtonItem button : TinkerStationButtonsWidget.this.buttons) {
+        button.pressed = false;
+      }
+      if (self instanceof SlotButtonItem slotInformationButton) {
+        slotInformationButton.pressed = true;
+        parent.onToolSelection(slotInformationButton.getLayout());
+      }
+    };
+
+    // repair button
+    SlotButtonItem slotButtonItem = new SlotButtonItem(0, -1, -1, parent.getDefaultLayout(), onButtonPressed);
+    this.addInfoButton(slotButtonItem);
+    if (parent.getDefaultLayout() == parent.getCurrentLayout()) {
+      slotButtonItem.pressed = true;
+    }
+
+    // tool buttons
+    int index = 1;
+    for (StationSlotLayout layout : StationSlotLayoutLoader.getInstance().getSortedSlots()) {
+      if (layout.getInputSlots().size() <= parent.getMaxInputs()) {
+        slotButtonItem = new SlotButtonItem(index, -1, -1, layout, onButtonPressed);
+        this.addInfoButton(slotButtonItem);
+        if (layout == parent.getCurrentLayout()) {
+          slotButtonItem.pressed = true;
+        }
+        index++;
+      }
+    }
+  }
+
+  public void addInfoButton(SlotButtonItem slotButtonItem) {
```

Comment:
While we are refactoring, do these functions need to be public? I find its best to use a minimal access level when possible. In general check if any of the functions in renamed classes can be given a lower access level (ignoring package, I tend to stick with public, private, and protected)

Would be nice to get javadocs on these if they need to be public, but I will not require that for the PR since they existed before.